### PR TITLE
PE-5575 Adding new role peak-user

### PR DESCRIPTION
**Pull request by ritikadhyawala**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: docker_init.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ superset db upgrade
 superset init
 
 flask fab create-user \
-  --role Gamma \
+  --role peak_user \
```
Comment (knowvishnu) [inline docker_init.sh]: please do not remove gamma role. and define peak_user role for new user 
```
flask fab create-user \
  --role Peakuser \
  --username peakuser \
  --firstname peakuser \
  --lastname peakuser \
  --email $PEAK_USER_EMAIL \
  --password $PEAK_USER_PASSWORD
```

Comment (knowvishnu) [reply docker_init.sh]: https://github.com/PeakBI/incubator-superset/pull/33/files

File: docker_init.sh [hunk 2]
```diff
@@ -42,6 +42,14 @@ flask fab create-user \
   --email $GUEST_EMAIL \
   --password $GUEST_PASSWORD
 
+flask fab create-user \
+  --role peak_user \
+  --username peakuser \
+  --firstname peakuser \
```
Comment (atom217) [inline docker_init.sh]: why we have same value for all three variables ?

Does it  gets reflected on front end as well ?

File: superset/security.py [hunk 3]
```diff
@@ -74,6 +74,8 @@ def __init__(self, **kwargs):
 class SupersetSecurityManager(SecurityManager):
     READ_ONLY_MODEL_VIEWS = {"DatabaseAsync", "DatabaseView", "DruidClusterModelView"}
 
+
+
```
Comment (atom217) [inline superset/security.py]: space not needed

File: docker_init.sh (hunk 2 above)
Comment (knowvishnu) [reply docker_init.sh]: yes  currently firstname and lastname will be same. it will rellect in superset profile

File: superset/custom_auth.py [hunk 4]
```diff
@@ -51,7 +57,9 @@ def wraps(self, *args, **kwargs):
                 user = 'admin'
             else:
                 privileges = loads(auth_response['privileges'])
-                if not has_resource_access(privileges):
+                if has_solution_write_access(privileges):
+                    user = 'peakuser'
+                elif not has_resource_access(privileges):
```
Comment (shubhamPeak) [inline superset/custom_auth.py]: I think this should be just else

File: superset/security.py [hunk 5]
```diff
@@ -82,6 +84,17 @@ class SupersetSecurityManager(SecurityManager):
         "UserRemoteUserModelView",
     }
 
+    GAMMA_MINUS_PEAK_USER_VIEW_MENUS = {
```
Comment (shubhamPeak) [inline superset/security.py]: it'll be MENUS or MINUS?? i think it's a spell mistake

File: superset/custom_auth.py (hunk 4 above)
Comment (knowvishnu) [reply superset/custom_auth.py]: no need to check user have customerai and demand ai permission or not.

File: superset/security.py (hunk 5 above)
Comment (knowvishnu) [reply superset/security.py]: its menu

Comment (shubhamPeak) [reply superset/security.py]: I'm saying about GAMMA_MINUS_PEAK_USER_VIEW_MENUS. IT'll be GAMMA_MENUS_PEAK_USER_VIEW_MENUS ??

File: superset/custom_auth.py (hunk 4 above)
Comment (shubhamPeak) [reply superset/custom_auth.py]: ok.. just a question. What will happen if the user doesn't have `write` access but have `read` access to the resource.
then it won't go in any of the if or elseIf cases.. In that scenario what should happen??

File: superset/security.py (hunk 5 above)
Comment (ritikadhyawala) [reply superset/security.py]: @knowvishnu @shubhamPeak It's minus only

File: superset/custom_auth.py (hunk 4 above)
Comment (knowvishnu) [reply superset/custom_auth.py]: in that scenario user will be guest user
https://github.com/PeakBI/incubator-superset/blob/master/superset/custom_auth.py#L38

Comment (knowvishnu) [reply superset/custom_auth.py]: but user must have the permission of cutomerai or demand ai

**Pull request merged**